### PR TITLE
Added support for testflight groups

### PR DESCRIPTION
**Pull request by taquitos**
Significant refactor, moved business logic to BoardingService
Moved most references to ENV[] into BoardingService and set at instantiation
Added startup checks to fail fast so you aren't left with an error'ing page if groups don't exist

### CODE REVIEW

File: app/services/boarding_service.rb [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      Rails.logger.info "Done"
+    end
+    return add_tester_response
+  end
+
+  private
+    def ensure_values
+      raise "Environment variable `ITC_APP_ID` required" if @app_id.empty?
+      raise "Environment variable `ITC_USER` or `FASTLANE_USER` required" if @user.empty?
+      raise "Environment variable `ITC_PASSWORD` or `FASTLANE_PASSWORD`" if @password.empty?
```
Comment (ohayon) [inline app/services/boarding_service.rb]: Should we check all of these and concatenate a message that more than one could be missing before raising?

Something like:
```
error_message = ""
error_message += "Environment variable `ITC_APP_ID` required" if @app_id.empty?
error_message += "Environment variable `ITC_USER` or `FASTLANE_USER` required" if @app_id.empty?
...
raise error_message # or if we have access to UI here we should do a UI.user_error! call
```

Comment (KrauseFx) [inline app/services/boarding_service.rb]: Yeah, I like the idea, great feedback @ohayon. You can use of more Ruby things, which would also add a space or new line inbetween the error messages:

```ruby
error_message = []
error_message << "Environment variable `ITC_APP_ID` required" if @app_id.empty?
error_message << "Environment variable `ITC_USER` or `FASTLANE_USER` required" if @app_id.empty?
...
raise error_message.join("\n")
```

File: config/initializers/services.rb [hunk 2]
```diff
@@ -0,0 +1 @@
+BOARDING_SERVICE = BoardingService.new
```
Comment (taquitos) [inline config/initializers/services.rb]: Yeah, everything inside initializers is run during app startup, so by creating the service here, we get startup-time checking of all ENV values. That way we don't have to wait for page load to see if we forgot something

File: app/services/boarding_service.rb [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+    def add_tester_to_groups!(tester: nil, app: nil, groups: nil)
+      if groups.nil?
+          default_external_group = self.default_external_group
+          if default_external_group.nil?
+            Rails.logger.error "The app #{self.name} does not have a default external group. Please make sure to pass group names to the `:groups` option."
+          end
+          test_flight_groups = [default_external_group]
+      else
+        test_flight_groups = Spaceship::TestFlight::Group.filter_groups(app_id: app.apple_id) do |group| groups.include?(group.name) end
```
Comment (taquitos) [inline app/services/boarding_service.rb]: 👍

File: app/services/boarding_service.rb (hunk 1 above)
Comment (taquitos) [inline app/services/boarding_service.rb]: Love this idea

**Comment by ohayon** [comment]
This is a great refactor!!

**Comment by ohayon** [comment]
🚀

**Pull request merged**